src/handlers/actions/moveFunds.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  AnyColonyClient,
+  ColonyClientV1,
+  ColonyClientV2,
+  ColonyClientV3,
+  ColonyClientV4,
+} from '@colony/colony-js';
+import { BigNumber } from 'ethers';
+import networkClient from '~/networkClient';
+import { ColonyActionType, ContractEvent } from '~/types';
+import { toNumber, writeActionFromEvent } from '~/utils';
+import { getDatabaseDomainId } from '~/utils/domains';
+
+/**
+ * The handler makes use of colonyClient getDomainFromFundingPot method which is only
+ * available on ColonyClientV5 and above. The following type predicate allows to check
+ * we're dealing with a client that supports this method
+ */
+type SupportedColonyClient = Exclude<
+  AnyColonyClient,
+  ColonyClientV1 | ColonyClientV2 | ColonyClientV3 | ColonyClientV4
+>;
+const isSupportedColonyClient = (
+  colonyClient: AnyColonyClient,
+): colonyClient is SupportedColonyClient =>
+  (colonyClient as SupportedColonyClient).getDomainFromFundingPot !== undefined;
```

Comment:
alternatively: 

```js
'getDomainFromFundingPot' in colonyClient
```

Then you don't need to cast `colonyClient`

Reply:
I think explicit casting might be better here since it will alert you if the property you're checking against doesn't exist/got removed
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates